src/reducers/vis-state-updaters.js

```diff
@@ -937,6 +940,69 @@ export const removeLayerUpdater = (state, {idx}) => {
   return updateAnimationDomain(newState);
 };
 
+/**
+ * duplicate layer
+ * @memberof visStateUpdaters
+ * @type {typeof import('./vis-state-updaters').duplicateLayerUpdater}
+ * @public
+ */
+export const duplicateLayerUpdater = (state, {idx}) => {
+  const {layers} = state;
+  const original = state.layers[idx];
+  if (!original) {
+    Console.warn(`layer.${idx} is undefined`);
+    return state;
+  }
+  let newLabel = `Copy of ${original.config.label}`;
+  let postfix = 0;
+  while (layers.find(l => l.config.label === newLabel)) {
+    newLabel = `${newLabel} ${postfix++}`;
```

Comment:
This will probably generate something like `Copy of My layer 0 1 2 3 4 5 6` and so on: starting with 0 and appending the next number to it. It'd be probably better to define some kind of template in a separate constant like `Copy of ${layerName}${postfix}` and then replace the variables in the loop.

Reply:
good catch